crates/net/network/tests/it/connect.rs

```diff
@@ -551,3 +551,34 @@ async fn test_shutdown() {
     let (_peer, reason) = listener0.next_session_closed().await.unwrap();
     assert_eq!(reason, Some(DisconnectReason::DisconnectRequested));
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_disconnect_incoming_when_exceeded_incoming_connections() {
+    let mut net = Testnet::create(1).await;
+    let (reth_p2p, reth_disc) = unused_tcp_udp();
+    let secret_key = SecretKey::new(&mut rand::thread_rng());
+    let peers_config = PeersConfig::default().with_max_inbound(0);
+
+    let config = NetworkConfigBuilder::new(secret_key)
+        .listener_addr(reth_p2p)
+        .discovery_addr(reth_disc)
+        .peer_config(peers_config)
+        .build(Arc::new(NoopProvider::default()));
+    let network = NetworkManager::new(config).await.unwrap();
+
+    let mut other_peer_handle = net.handles().next().unwrap();
+
+    let handle = network.handle().clone();
+
+    other_peer_handle.add_peer(*handle.peer_id(), handle.local_addr());
+
+    tokio::task::spawn(network);
+    let _handle = net.spawn();
+
+    // create NetworkEventStream to get the next session established event easily
+    let mut events = other_peer_handle.event_listener();
+
+    while let ev = events.next().await {
+        println!("{ev:?}");
+    }
```

Comment:
ah yes, this won't work because there's no event here for failed connection attempt, it would be too noisy.

Instead we should verify this by
* sleep for a bit
* check number of connected peers is 0

we eventually need additional event channels for `AdminAPI` events but for now this should be fine.
I've checked via tracing logs that this works as expected.